internal/types/document_test.go

```diff
@@ -75,23 +75,31 @@ func TestDocument(t *testing.T) {
 		assert.Equal(t, int32(42), must.NotFail(b.Get("foo")))
 	})
 
-	t.Run("SetID", func(t *testing.T) {
+	t.Run("MoveIDToTheFirstIndex", func(t *testing.T) {
 		t.Parallel()
 
 		doc := must.NotFail(NewDocument(
 			"_id", int32(42),
 			"foo", "bar",
+			"baz", "qux",
 		))
-		assert.Equal(t, []string{"_id", "foo"}, doc.Keys())
+		assert.Equal(t, []string{"_id", "foo", "baz"}, doc.Keys())
 
 		doc = must.NotFail(NewDocument(
 			"foo", "bar",
 			"_id", int32(42),
+			"baz", "qux",
 		))
-		assert.Equal(t, []string{"_id", "foo"}, doc.Keys())
+		doc.MoveIDToTheFirstIndex()
+		assert.Equal(t, []string{"_id", "foo", "baz"}, doc.Keys())
 
-		doc.Set("_id", "bar")
-		assert.Equal(t, []string{"_id", "foo"}, doc.Keys())
+		doc = must.NotFail(NewDocument(
+			"foo", "bar",
+			"baz", "qux",
+			"_id", int32(42),
+		))
+		doc.MoveIDToTheFirstIndex()
+		assert.Equal(t, []string{"_id", "foo", "baz"}, doc.Keys())
 	})
```

Comment:
I like that we have tests for different options! Let's also add one for the case when `_id` is not present in the doc at all? So then we expect nothing is changed in the doc (but nothing panics etc). Codecov currently shows this case uncovered.

Reply:
Done.